### PR TITLE
Improve completions for union field patterns

### PR DESCRIPTION
**Pull request by kerams**
... when no identifier has been typed yet and the user presses Ctrl-space. In this case, the original ordering of completions from FCS is preserved, so we get suggested names at the top of the list (and the list itself is filtered). If you instead trigger completions by typing the first letter, VS will partially sort the list. That's not great when you don't necessarily know what you're looking for, and the suggested names would have been helpful.

Before

devenv_gFrkZZaZFw

Current, with the first letter, sorted by VS

devenv_Kda9BH6crJ

After

devenv_tz98OS9mh2
devenv_wn27c9fRqx

### CODE REVIEW

File: src/Compiler/Service/ServiceParsedInputOps.fs [hunk 1]
```diff
@@ -1312,7 +1330,7 @@ module ParsedInput =
             TryGetCompletionContextInPattern suppressIdentifierCompletions pat1 None pos
             |> Option.orElseWith (fun () -> TryGetCompletionContextInPattern suppressIdentifierCompletions pat2 None pos)
         | SynPat.IsInst (_, m) when rangeContainsPos m pos -> Some CompletionContext.Type
-        | SynPat.Wild m when rangeContainsPos m pos -> Some CompletionContext.Invalid
+        | SynPat.Wild m when rangeContainsPos m pos && m.StartColumn <> m.EndColumn -> Some CompletionContext.Invalid
```
Comment (psfinaki) [inline src/Compiler/Service/ServiceParsedInputOps.fs:1333]: Just curious, why is this needed? When would this come into play?

**Review by psfinaki — APPROVED** [review]
Good stuff as usual, thanks @kerams :)

I am thinking we might put all these recent changes together in a blog post, that's a substantial improvement of the UX.

File: src/Compiler/Service/ServiceParsedInputOps.fs (hunk 1 above)
Comment (psfinaki) [reply src/Compiler/Service/ServiceParsedInputOps.fs:1333]: Hmm good to know, thanks!

Comment (auduchinok) [reply src/Compiler/Service/ServiceParsedInputOps.fs:1333]: > They're not even wrapped in SynPat.FromParserError.

We should probably either wrap them or add a union case that represents an error at a range without including another pattern inside of it, similar to `SynExpr.ArbitraryAfterError`.

Comment (kerams) [reply src/Compiler/Service/ServiceParsedInputOps.fs:1333]: @nojaf, do you have an opinion on this?

Comment (psfinaki) [reply src/Compiler/Service/ServiceParsedInputOps.fs:1333]: Side note, nojaf is on vacation, it might take them long time to respond. :)

Comment (kerams) [reply src/Compiler/Service/ServiceParsedInputOps.fs:1333]: Alright, so do we merge this and return to revamp the represantation of errors in patterns later?

Comment (psfinaki) [reply src/Compiler/Service/ServiceParsedInputOps.fs:1333]: Yes, I don't mind merging it this way.

Comment (kerams) [reply src/Compiler/Service/ServiceParsedInputOps.fs:1333]: The top 2 completions are [suggested names](https://github.com/dotnet/fsharp/pull/15530).

Comment (nojaf) [reply src/Compiler/Service/ServiceParsedInputOps.fs:1333]: Oh I see.
